src/components/panel/panel.js

```diff
@@ -885,42 +858,38 @@ MdPanelRef.prototype.close = function() {
  * the panel is attached.
  */
 MdPanelRef.prototype.attach = function() {
-  if (this.isAttached) {
-    return this._attachPromise;
+  if (this.isAttached && this._panelEl) {
+    return this._$q.when(this);
   }
 
-  // TODO(ErinCoughlan) - Cancel any in-progress actions.
-
   var self = this;
-
-  this._attachPromise = this._$q.all([
-    this._createBackdrop(),
-    this._createPanel().then(function() {
-      self.isAttached = true;
-      self._addEventListeners();
-      return self._$q.resolve(self);
-    }, this._$q.reject)
-  ]);
-
-  return this._attachPromise;
+  return this._$q(function(resolve, reject) {
+    var done = self._done(resolve, self);
+
+    self._$q.all([
+      self._createBackdrop(),
+      self._createPanel().then(function() {
+        self.isAttached = true;
+        self._addEventListeners();
+      }, reject)
+    ]).then(done)
+        .catch(reject);
```

Comment:
Do you expect `done()` to reject or throw an exception? 
* If yes, the `.catch(reject)` is good. 
* If no, then you can do `.then( done, reject )`